// @aws-sdk/client-greengrass/dist/es/models/models_0.js
import { __assign } from "tslib";
export var ConnectivityInfo;
(function (ConnectivityInfo) {
    ConnectivityInfo.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ConnectivityInfo || (ConnectivityInfo = {}));
export var Connector;
(function (Connector) {
    Connector.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(Connector || (Connector = {}));
export var Core;
(function (Core) {
    Core.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(Core || (Core = {}));
export var DefinitionInformation;
(function (DefinitionInformation) {
    DefinitionInformation.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DefinitionInformation || (DefinitionInformation = {}));
export var Device;
(function (Device) {
    Device.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(Device || (Device = {}));
export var EncodingType;
(function (EncodingType) {
    EncodingType["binary"] = "binary";
    EncodingType["json"] = "json";
})(EncodingType || (EncodingType = {}));
export var FunctionIsolationMode;
(function (FunctionIsolationMode) {
    FunctionIsolationMode["GreengrassContainer"] = "GreengrassContainer";
    FunctionIsolationMode["NoContainer"] = "NoContainer";
})(FunctionIsolationMode || (FunctionIsolationMode = {}));
export var FunctionRunAsConfig;
(function (FunctionRunAsConfig) {
    FunctionRunAsConfig.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(FunctionRunAsConfig || (FunctionRunAsConfig = {}));
export var FunctionExecutionConfig;
(function (FunctionExecutionConfig) {
    FunctionExecutionConfig.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(FunctionExecutionConfig || (FunctionExecutionConfig = {}));
export var Permission;
(function (Permission) {
    Permission["ro"] = "ro";
    Permission["rw"] = "rw";
})(Permission || (Permission = {}));
export var ResourceAccessPolicy;
(function (ResourceAccessPolicy) {
    ResourceAccessPolicy.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ResourceAccessPolicy || (ResourceAccessPolicy = {}));
export var FunctionConfigurationEnvironment;
(function (FunctionConfigurationEnvironment) {
    FunctionConfigurationEnvironment.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(FunctionConfigurationEnvironment || (FunctionConfigurationEnvironment = {}));
export var FunctionConfiguration;
(function (FunctionConfiguration) {
    FunctionConfiguration.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(FunctionConfiguration || (FunctionConfiguration = {}));
export var Function;
(function (Function) {
    Function.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(Function || (Function = {}));
export var GroupCertificateAuthorityProperties;
(function (GroupCertificateAuthorityProperties) {
    GroupCertificateAuthorityProperties.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GroupCertificateAuthorityProperties || (GroupCertificateAuthorityProperties = {}));
export var GroupInformation;
(function (GroupInformation) {
    GroupInformation.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GroupInformation || (GroupInformation = {}));
export var LoggerComponent;
(function (LoggerComponent) {
    LoggerComponent["GreengrassSystem"] = "GreengrassSystem";
    LoggerComponent["Lambda"] = "Lambda";
})(LoggerComponent || (LoggerComponent = {}));
export var LoggerLevel;
(function (LoggerLevel) {
    LoggerLevel["DEBUG"] = "DEBUG";
    LoggerLevel["ERROR"] = "ERROR";
    LoggerLevel["FATAL"] = "FATAL";
    LoggerLevel["INFO"] = "INFO";
    LoggerLevel["WARN"] = "WARN";
})(LoggerLevel || (LoggerLevel = {}));
export var LoggerType;
(function (LoggerType) {
    LoggerType["AWSCloudWatch"] = "AWSCloudWatch";
    LoggerType["FileSystem"] = "FileSystem";
})(LoggerType || (LoggerType = {}));
export var Logger;
(function (Logger) {
    Logger.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(Logger || (Logger = {}));
export var GroupOwnerSetting;
(function (GroupOwnerSetting) {
    GroupOwnerSetting.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GroupOwnerSetting || (GroupOwnerSetting = {}));
export var LocalDeviceResourceData;
(function (LocalDeviceResourceData) {
    LocalDeviceResourceData.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(LocalDeviceResourceData || (LocalDeviceResourceData = {}));
export var LocalVolumeResourceData;
(function (LocalVolumeResourceData) {
    LocalVolumeResourceData.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(LocalVolumeResourceData || (LocalVolumeResourceData = {}));
export var ResourceDownloadOwnerSetting;
(function (ResourceDownloadOwnerSetting) {
    ResourceDownloadOwnerSetting.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ResourceDownloadOwnerSetting || (ResourceDownloadOwnerSetting = {}));
export var S3MachineLearningModelResourceData;
(function (S3MachineLearningModelResourceData) {
    S3MachineLearningModelResourceData.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(S3MachineLearningModelResourceData || (S3MachineLearningModelResourceData = {}));
export var SageMakerMachineLearningModelResourceData;
(function (SageMakerMachineLearningModelResourceData) {
    SageMakerMachineLearningModelResourceData.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(SageMakerMachineLearningModelResourceData || (SageMakerMachineLearningModelResourceData = {}));
export var SecretsManagerSecretResourceData;
(function (SecretsManagerSecretResourceData) {
    SecretsManagerSecretResourceData.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(SecretsManagerSecretResourceData || (SecretsManagerSecretResourceData = {}));
export var ResourceDataContainer;
(function (ResourceDataContainer) {
    ResourceDataContainer.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ResourceDataContainer || (ResourceDataContainer = {}));
export var Resource;
(function (Resource) {
    Resource.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(Resource || (Resource = {}));
export var Subscription;
(function (Subscription) {
    Subscription.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(Subscription || (Subscription = {}));
export var VersionInformation;
(function (VersionInformation) {
    VersionInformation.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(VersionInformation || (VersionInformation = {}));
export var AssociateRoleToGroupRequest;
(function (AssociateRoleToGroupRequest) {
    AssociateRoleToGroupRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AssociateRoleToGroupRequest || (AssociateRoleToGroupRequest = {}));
export var AssociateRoleToGroupResponse;
(function (AssociateRoleToGroupResponse) {
    AssociateRoleToGroupResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AssociateRoleToGroupResponse || (AssociateRoleToGroupResponse = {}));
export var ErrorDetail;
(function (ErrorDetail) {
    ErrorDetail.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ErrorDetail || (ErrorDetail = {}));
export var BadRequestException;
(function (BadRequestException) {
    BadRequestException.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(BadRequestException || (BadRequestException = {}));
export var InternalServerErrorException;
(function (InternalServerErrorException) {
    InternalServerErrorException.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(InternalServerErrorException || (InternalServerErrorException = {}));
export var AssociateServiceRoleToAccountRequest;
(function (AssociateServiceRoleToAccountRequest) {
    AssociateServiceRoleToAccountRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AssociateServiceRoleToAccountRequest || (AssociateServiceRoleToAccountRequest = {}));
export var AssociateServiceRoleToAccountResponse;
(function (AssociateServiceRoleToAccountResponse) {
    AssociateServiceRoleToAccountResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AssociateServiceRoleToAccountResponse || (AssociateServiceRoleToAccountResponse = {}));
export var BulkDeployment;
(function (BulkDeployment) {
    BulkDeployment.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(BulkDeployment || (BulkDeployment = {}));
export var BulkDeploymentMetrics;
(function (BulkDeploymentMetrics) {
    BulkDeploymentMetrics.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(BulkDeploymentMetrics || (BulkDeploymentMetrics = {}));
export var DeploymentType;
(function (DeploymentType) {
    DeploymentType["ForceResetDeployment"] = "ForceResetDeployment";
    DeploymentType["NewDeployment"] = "NewDeployment";
    DeploymentType["Redeployment"] = "Redeployment";
    DeploymentType["ResetDeployment"] = "ResetDeployment";
})(DeploymentType || (DeploymentType = {}));
export var BulkDeploymentResult;
(function (BulkDeploymentResult) {
    BulkDeploymentResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(BulkDeploymentResult || (BulkDeploymentResult = {}));
export var BulkDeploymentStatus;
(function (BulkDeploymentStatus) {
    BulkDeploymentStatus["Completed"] = "Completed";
    BulkDeploymentStatus["Failed"] = "Failed";
    BulkDeploymentStatus["Initializing"] = "Initializing";
    BulkDeploymentStatus["Running"] = "Running";
    BulkDeploymentStatus["Stopped"] = "Stopped";
    BulkDeploymentStatus["Stopping"] = "Stopping";
})(BulkDeploymentStatus || (BulkDeploymentStatus = {}));
export var ConfigurationSyncStatus;
(function (ConfigurationSyncStatus) {
    ConfigurationSyncStatus["InSync"] = "InSync";
    ConfigurationSyncStatus["OutOfSync"] = "OutOfSync";
})(ConfigurationSyncStatus || (ConfigurationSyncStatus = {}));
export var ConnectorDefinitionVersion;
(function (ConnectorDefinitionVersion) {
    ConnectorDefinitionVersion.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ConnectorDefinitionVersion || (ConnectorDefinitionVersion = {}));
export var CoreDefinitionVersion;
(function (CoreDefinitionVersion) {
    CoreDefinitionVersion.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CoreDefinitionVersion || (CoreDefinitionVersion = {}));
export var CreateConnectorDefinitionRequest;
(function (CreateConnectorDefinitionRequest) {
    CreateConnectorDefinitionRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateConnectorDefinitionRequest || (CreateConnectorDefinitionRequest = {}));
export var CreateConnectorDefinitionResponse;
(function (CreateConnectorDefinitionResponse) {
    CreateConnectorDefinitionResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateConnectorDefinitionResponse || (CreateConnectorDefinitionResponse = {}));
export var CreateConnectorDefinitionVersionRequest;
(function (CreateConnectorDefinitionVersionRequest) {
    CreateConnectorDefinitionVersionRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateConnectorDefinitionVersionRequest || (CreateConnectorDefinitionVersionRequest = {}));
export var CreateConnectorDefinitionVersionResponse;
(function (CreateConnectorDefinitionVersionResponse) {
    CreateConnectorDefinitionVersionResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateConnectorDefinitionVersionResponse || (CreateConnectorDefinitionVersionResponse = {}));
export var CreateCoreDefinitionRequest;
(function (CreateCoreDefinitionRequest) {
    CreateCoreDefinitionRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateCoreDefinitionRequest || (CreateCoreDefinitionRequest = {}));
export var CreateCoreDefinitionResponse;
(function (CreateCoreDefinitionResponse) {
    CreateCoreDefinitionResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateCoreDefinitionResponse || (CreateCoreDefinitionResponse = {}));
export var CreateCoreDefinitionVersionRequest;
(function (CreateCoreDefinitionVersionRequest) {
    CreateCoreDefinitionVersionRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateCoreDefinitionVersionRequest || (CreateCoreDefinitionVersionRequest = {}));
export var CreateCoreDefinitionVersionResponse;
(function (CreateCoreDefinitionVersionResponse) {
    CreateCoreDefinitionVersionResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateCoreDefinitionVersionResponse || (CreateCoreDefinitionVersionResponse = {}));
export var CreateDeploymentRequest;
(function (CreateDeploymentRequest) {
    CreateDeploymentRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateDeploymentRequest || (CreateDeploymentRequest = {}));
export var CreateDeploymentResponse;
(function (CreateDeploymentResponse) {
    CreateDeploymentResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateDeploymentResponse || (CreateDeploymentResponse = {}));
export var DeviceDefinitionVersion;
(function (DeviceDefinitionVersion) {
    DeviceDefinitionVersion.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeviceDefinitionVersion || (DeviceDefinitionVersion = {}));
export var CreateDeviceDefinitionRequest;
(function (CreateDeviceDefinitionRequest) {
    CreateDeviceDefinitionRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateDeviceDefinitionRequest || (CreateDeviceDefinitionRequest = {}));
export var CreateDeviceDefinitionResponse;
(function (CreateDeviceDefinitionResponse) {
    CreateDeviceDefinitionResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateDeviceDefinitionResponse || (CreateDeviceDefinitionResponse = {}));
export var CreateDeviceDefinitionVersionRequest;
(function (CreateDeviceDefinitionVersionRequest) {
    CreateDeviceDefinitionVersionRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateDeviceDefinitionVersionRequest || (CreateDeviceDefinitionVersionRequest = {}));
export var CreateDeviceDefinitionVersionResponse;
(function (CreateDeviceDefinitionVersionResponse) {
    CreateDeviceDefinitionVersionResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateDeviceDefinitionVersionResponse || (CreateDeviceDefinitionVersionResponse = {}));
export var FunctionDefaultExecutionConfig;
(function (FunctionDefaultExecutionConfig) {
    FunctionDefaultExecutionConfig.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(FunctionDefaultExecutionConfig || (FunctionDefaultExecutionConfig = {}));
export var FunctionDefaultConfig;
(function (FunctionDefaultConfig) {
    FunctionDefaultConfig.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(FunctionDefaultConfig || (FunctionDefaultConfig = {}));
export var FunctionDefinitionVersion;
(function (FunctionDefinitionVersion) {
    FunctionDefinitionVersion.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(FunctionDefinitionVersion || (FunctionDefinitionVersion = {}));
export var CreateFunctionDefinitionRequest;
(function (CreateFunctionDefinitionRequest) {
    CreateFunctionDefinitionRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateFunctionDefinitionRequest || (CreateFunctionDefinitionRequest = {}));
export var CreateFunctionDefinitionResponse;
(function (CreateFunctionDefinitionResponse) {
    CreateFunctionDefinitionResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateFunctionDefinitionResponse || (CreateFunctionDefinitionResponse = {}));
export var CreateFunctionDefinitionVersionRequest;
(function (CreateFunctionDefinitionVersionRequest) {
    CreateFunctionDefinitionVersionRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateFunctionDefinitionVersionRequest || (CreateFunctionDefinitionVersionRequest = {}));
export var CreateFunctionDefinitionVersionResponse;
(function (CreateFunctionDefinitionVersionResponse) {
    CreateFunctionDefinitionVersionResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateFunctionDefinitionVersionResponse || (CreateFunctionDefinitionVersionResponse = {}));
export var GroupVersion;
(function (GroupVersion) {
    GroupVersion.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GroupVersion || (GroupVersion = {}));
export var CreateGroupRequest;
(function (CreateGroupRequest) {
    CreateGroupRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateGroupRequest || (CreateGroupRequest = {}));
export var CreateGroupResponse;
(function (CreateGroupResponse) {
    CreateGroupResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateGroupResponse || (CreateGroupResponse = {}));
export var CreateGroupCertificateAuthorityRequest;
(function (CreateGroupCertificateAuthorityRequest) {
    CreateGroupCertificateAuthorityRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateGroupCertificateAuthorityRequest || (CreateGroupCertificateAuthorityRequest = {}));
export var CreateGroupCertificateAuthorityResponse;
(function (CreateGroupCertificateAuthorityResponse) {
    CreateGroupCertificateAuthorityResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateGroupCertificateAuthorityResponse || (CreateGroupCertificateAuthorityResponse = {}));
export var CreateGroupVersionRequest;
(function (CreateGroupVersionRequest) {
    CreateGroupVersionRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateGroupVersionRequest || (CreateGroupVersionRequest = {}));
export var CreateGroupVersionResponse;
(function (CreateGroupVersionResponse) {
    CreateGroupVersionResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateGroupVersionResponse || (CreateGroupVersionResponse = {}));
export var LoggerDefinitionVersion;
(function (LoggerDefinitionVersion) {
    LoggerDefinitionVersion.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(LoggerDefinitionVersion || (LoggerDefinitionVersion = {}));
export var CreateLoggerDefinitionRequest;
(function (CreateLoggerDefinitionRequest) {
    CreateLoggerDefinitionRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateLoggerDefinitionRequest || (CreateLoggerDefinitionRequest = {}));
export var CreateLoggerDefinitionResponse;
(function (CreateLoggerDefinitionResponse) {
    CreateLoggerDefinitionResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateLoggerDefinitionResponse || (CreateLoggerDefinitionResponse = {}));
export var CreateLoggerDefinitionVersionRequest;
(function (CreateLoggerDefinitionVersionRequest) {
    CreateLoggerDefinitionVersionRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateLoggerDefinitionVersionRequest || (CreateLoggerDefinitionVersionRequest = {}));
export var CreateLoggerDefinitionVersionResponse;
(function (CreateLoggerDefinitionVersionResponse) {
    CreateLoggerDefinitionVersionResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateLoggerDefinitionVersionResponse || (CreateLoggerDefinitionVersionResponse = {}));
export var ResourceDefinitionVersion;
(function (ResourceDefinitionVersion) {
    ResourceDefinitionVersion.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ResourceDefinitionVersion || (ResourceDefinitionVersion = {}));
export var CreateResourceDefinitionRequest;
(function (CreateResourceDefinitionRequest) {
    CreateResourceDefinitionRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateResourceDefinitionRequest || (CreateResourceDefinitionRequest = {}));
export var CreateResourceDefinitionResponse;
(function (CreateResourceDefinitionResponse) {
    CreateResourceDefinitionResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateResourceDefinitionResponse || (CreateResourceDefinitionResponse = {}));
export var CreateResourceDefinitionVersionRequest;
(function (CreateResourceDefinitionVersionRequest) {
    CreateResourceDefinitionVersionRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateResourceDefinitionVersionRequest || (CreateResourceDefinitionVersionRequest = {}));
export var CreateResourceDefinitionVersionResponse;
(function (CreateResourceDefinitionVersionResponse) {
    CreateResourceDefinitionVersionResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateResourceDefinitionVersionResponse || (CreateResourceDefinitionVersionResponse = {}));
export var SoftwareToUpdate;
(function (SoftwareToUpdate) {
    SoftwareToUpdate["core"] = "core";
    SoftwareToUpdate["ota_agent"] = "ota_agent";
})(SoftwareToUpdate || (SoftwareToUpdate = {}));
export var UpdateAgentLogLevel;
(function (UpdateAgentLogLevel) {
    UpdateAgentLogLevel["DEBUG"] = "DEBUG";
    UpdateAgentLogLevel["ERROR"] = "ERROR";
    UpdateAgentLogLevel["FATAL"] = "FATAL";
    UpdateAgentLogLevel["INFO"] = "INFO";
    UpdateAgentLogLevel["NONE"] = "NONE";
    UpdateAgentLogLevel["TRACE"] = "TRACE";
    UpdateAgentLogLevel["VERBOSE"] = "VERBOSE";
    UpdateAgentLogLevel["WARN"] = "WARN";
})(UpdateAgentLogLevel || (UpdateAgentLogLevel = {}));
export var UpdateTargetsArchitecture;
(function (UpdateTargetsArchitecture) {
    UpdateTargetsArchitecture["aarch64"] = "aarch64";
    UpdateTargetsArchitecture["armv6l"] = "armv6l";
    UpdateTargetsArchitecture["armv7l"] = "armv7l";
    UpdateTargetsArchitecture["x86_64"] = "x86_64";
})(UpdateTargetsArchitecture || (UpdateTargetsArchitecture = {}));
export var UpdateTargetsOperatingSystem;
(function (UpdateTargetsOperatingSystem) {
    UpdateTargetsOperatingSystem["amazon_linux"] = "amazon_linux";
    UpdateTargetsOperatingSystem["openwrt"] = "openwrt";
    UpdateTargetsOperatingSystem["raspbian"] = "raspbian";
    UpdateTargetsOperatingSystem["ubuntu"] = "ubuntu";
})(UpdateTargetsOperatingSystem || (UpdateTargetsOperatingSystem = {}));
export var CreateSoftwareUpdateJobRequest;
(function (CreateSoftwareUpdateJobRequest) {
    CreateSoftwareUpdateJobRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateSoftwareUpdateJobRequest || (CreateSoftwareUpdateJobRequest = {}));
export var CreateSoftwareUpdateJobResponse;
(function (CreateSoftwareUpdateJobResponse) {
    CreateSoftwareUpdateJobResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateSoftwareUpdateJobResponse || (CreateSoftwareUpdateJobResponse = {}));
export var SubscriptionDefinitionVersion;
(function (SubscriptionDefinitionVersion) {
    SubscriptionDefinitionVersion.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(SubscriptionDefinitionVersion || (SubscriptionDefinitionVersion = {}));
export var CreateSubscriptionDefinitionRequest;
(function (CreateSubscriptionDefinitionRequest) {
    CreateSubscriptionDefinitionRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateSubscriptionDefinitionRequest || (CreateSubscriptionDefinitionRequest = {}));
export var CreateSubscriptionDefinitionResponse;
(function (CreateSubscriptionDefinitionResponse) {
    CreateSubscriptionDefinitionResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateSubscriptionDefinitionResponse || (CreateSubscriptionDefinitionResponse = {}));
export var CreateSubscriptionDefinitionVersionRequest;
(function (CreateSubscriptionDefinitionVersionRequest) {
    CreateSubscriptionDefinitionVersionRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateSubscriptionDefinitionVersionRequest || (CreateSubscriptionDefinitionVersionRequest = {}));
export var CreateSubscriptionDefinitionVersionResponse;
(function (CreateSubscriptionDefinitionVersionResponse) {
    CreateSubscriptionDefinitionVersionResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateSubscriptionDefinitionVersionResponse || (CreateSubscriptionDefinitionVersionResponse = {}));
export var DeleteConnectorDefinitionRequest;
(function (DeleteConnectorDefinitionRequest) {
    DeleteConnectorDefinitionRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteConnectorDefinitionRequest || (DeleteConnectorDefinitionRequest = {}));
export var DeleteConnectorDefinitionResponse;
(function (DeleteConnectorDefinitionResponse) {
    DeleteConnectorDefinitionResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteConnectorDefinitionResponse || (DeleteConnectorDefinitionResponse = {}));
export var DeleteCoreDefinitionRequest;
(function (DeleteCoreDefinitionRequest) {
    DeleteCoreDefinitionRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteCoreDefinitionRequest || (DeleteCoreDefinitionRequest = {}));
export var DeleteCoreDefinitionResponse;
(function (DeleteCoreDefinitionResponse) {
    DeleteCoreDefinitionResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteCoreDefinitionResponse || (DeleteCoreDefinitionResponse = {}));
export var DeleteDeviceDefinitionRequest;
(function (DeleteDeviceDefinitionRequest) {
    DeleteDeviceDefinitionRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteDeviceDefinitionRequest || (DeleteDeviceDefinitionRequest = {}));
export var DeleteDeviceDefinitionResponse;
(function (DeleteDeviceDefinitionResponse) {
    DeleteDeviceDefinitionResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteDeviceDefinitionResponse || (DeleteDeviceDefinitionResponse = {}));
export var DeleteFunctionDefinitionRequest;
(function (DeleteFunctionDefinitionRequest) {
    DeleteFunctionDefinitionRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteFunctionDefinitionRequest || (DeleteFunctionDefinitionRequest = {}));
export var DeleteFunctionDefinitionResponse;
(function (DeleteFunctionDefinitionResponse) {
    DeleteFunctionDefinitionResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteFunctionDefinitionResponse || (DeleteFunctionDefinitionResponse = {}));
export var DeleteGroupRequest;
(function (DeleteGroupRequest) {
    DeleteGroupRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteGroupRequest || (DeleteGroupRequest = {}));
export var DeleteGroupResponse;
(function (DeleteGroupResponse) {
    DeleteGroupResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteGroupResponse || (DeleteGroupResponse = {}));
export var DeleteLoggerDefinitionRequest;
(function (DeleteLoggerDefinitionRequest) {
    DeleteLoggerDefinitionRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteLoggerDefinitionRequest || (DeleteLoggerDefinitionRequest = {}));
export var DeleteLoggerDefinitionResponse;
(function (DeleteLoggerDefinitionResponse) {
    DeleteLoggerDefinitionResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteLoggerDefinitionResponse || (DeleteLoggerDefinitionResponse = {}));
export var DeleteResourceDefinitionRequest;
(function (DeleteResourceDefinitionRequest) {
    DeleteResourceDefinitionRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteResourceDefinitionRequest || (DeleteResourceDefinitionRequest = {}));
export var DeleteResourceDefinitionResponse;
(function (DeleteResourceDefinitionResponse) {
    DeleteResourceDefinitionResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteResourceDefinitionResponse || (DeleteResourceDefinitionResponse = {}));
export var DeleteSubscriptionDefinitionRequest;
(function (DeleteSubscriptionDefinitionRequest) {
    DeleteSubscriptionDefinitionRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteSubscriptionDefinitionRequest || (DeleteSubscriptionDefinitionRequest = {}));
export var DeleteSubscriptionDefinitionResponse;
(function (DeleteSubscriptionDefinitionResponse) {
    DeleteSubscriptionDefinitionResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteSubscriptionDefinitionResponse || (DeleteSubscriptionDefinitionResponse = {}));
export var Deployment;
(function (Deployment) {
    Deployment.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(Deployment || (Deployment = {}));
export var DisassociateRoleFromGroupRequest;
(function (DisassociateRoleFromGroupRequest) {
    DisassociateRoleFromGroupRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DisassociateRoleFromGroupRequest || (DisassociateRoleFromGroupRequest = {}));
export var DisassociateRoleFromGroupResponse;
(function (DisassociateRoleFromGroupResponse) {
    DisassociateRoleFromGroupResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DisassociateRoleFromGroupResponse || (DisassociateRoleFromGroupResponse = {}));
export var DisassociateServiceRoleFromAccountRequest;
(function (DisassociateServiceRoleFromAccountRequest) {
    DisassociateServiceRoleFromAccountRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DisassociateServiceRoleFromAccountRequest || (DisassociateServiceRoleFromAccountRequest = {}));
export var DisassociateServiceRoleFromAccountResponse;
(function (DisassociateServiceRoleFromAccountResponse) {
    DisassociateServiceRoleFromAccountResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DisassociateServiceRoleFromAccountResponse || (DisassociateServiceRoleFromAccountResponse = {}));
export var GetAssociatedRoleRequest;
(function (GetAssociatedRoleRequest) {
    GetAssociatedRoleRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetAssociatedRoleRequest || (GetAssociatedRoleRequest = {}));
export var GetAssociatedRoleResponse;
(function (GetAssociatedRoleResponse) {
    GetAssociatedRoleResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetAssociatedRoleResponse || (GetAssociatedRoleResponse = {}));
export var GetBulkDeploymentStatusRequest;
(function (GetBulkDeploymentStatusRequest) {
    GetBulkDeploymentStatusRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetBulkDeploymentStatusRequest || (GetBulkDeploymentStatusRequest = {}));
export var GetBulkDeploymentStatusResponse;
(function (GetBulkDeploymentStatusResponse) {
    GetBulkDeploymentStatusResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetBulkDeploymentStatusResponse || (GetBulkDeploymentStatusResponse = {}));
export var GetConnectivityInfoRequest;
(function (GetConnectivityInfoRequest) {
    GetConnectivityInfoRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetConnectivityInfoRequest || (GetConnectivityInfoRequest = {}));
export var GetConnectivityInfoResponse;
(function (GetConnectivityInfoResponse) {
    GetConnectivityInfoResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetConnectivityInfoResponse || (GetConnectivityInfoResponse = {}));
export var GetConnectorDefinitionRequest;
(function (GetConnectorDefinitionRequest) {
    GetConnectorDefinitionRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetConnectorDefinitionRequest || (GetConnectorDefinitionRequest = {}));
export var GetConnectorDefinitionResponse;
(function (GetConnectorDefinitionResponse) {
    GetConnectorDefinitionResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetConnectorDefinitionResponse || (GetConnectorDefinitionResponse = {}));
export var GetConnectorDefinitionVersionRequest;
(function (GetConnectorDefinitionVersionRequest) {
    GetConnectorDefinitionVersionRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetConnectorDefinitionVersionRequest || (GetConnectorDefinitionVersionRequest = {}));
export var GetConnectorDefinitionVersionResponse;
(function (GetConnectorDefinitionVersionResponse) {
    GetConnectorDefinitionVersionResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetConnectorDefinitionVersionResponse || (GetConnectorDefinitionVersionResponse = {}));
export var GetCoreDefinitionRequest;
(function (GetCoreDefinitionRequest) {
    GetCoreDefinitionRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetCoreDefinitionRequest || (GetCoreDefinitionRequest = {}));
export var GetCoreDefinitionResponse;
(function (GetCoreDefinitionResponse) {
    GetCoreDefinitionResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetCoreDefinitionResponse || (GetCoreDefinitionResponse = {}));
export var GetCoreDefinitionVersionRequest;
(function (GetCoreDefinitionVersionRequest) {
    GetCoreDefinitionVersionRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetCoreDefinitionVersionRequest || (GetCoreDefinitionVersionRequest = {}));
export var GetCoreDefinitionVersionResponse;
(function (GetCoreDefinitionVersionResponse) {
    GetCoreDefinitionVersionResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetCoreDefinitionVersionResponse || (GetCoreDefinitionVersionResponse = {}));
export var GetDeploymentStatusRequest;
(function (GetDeploymentStatusRequest) {
    GetDeploymentStatusRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetDeploymentStatusRequest || (GetDeploymentStatusRequest = {}));
export var GetDeploymentStatusResponse;
(function (GetDeploymentStatusResponse) {
    GetDeploymentStatusResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetDeploymentStatusResponse || (GetDeploymentStatusResponse = {}));
export var GetDeviceDefinitionRequest;
(function (GetDeviceDefinitionRequest) {
    GetDeviceDefinitionRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetDeviceDefinitionRequest || (GetDeviceDefinitionRequest = {}));
export var GetDeviceDefinitionResponse;
(function (GetDeviceDefinitionResponse) {
    GetDeviceDefinitionResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetDeviceDefinitionResponse || (GetDeviceDefinitionResponse = {}));
export var GetDeviceDefinitionVersionRequest;
(function (GetDeviceDefinitionVersionRequest) {
    GetDeviceDefinitionVersionRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetDeviceDefinitionVersionRequest || (GetDeviceDefinitionVersionRequest = {}));
export var GetDeviceDefinitionVersionResponse;
(function (GetDeviceDefinitionVersionResponse) {
    GetDeviceDefinitionVersionResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetDeviceDefinitionVersionResponse || (GetDeviceDefinitionVersionResponse = {}));
export var GetFunctionDefinitionRequest;
(function (GetFunctionDefinitionRequest) {
    GetFunctionDefinitionRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetFunctionDefinitionRequest || (GetFunctionDefinitionRequest = {}));
export var GetFunctionDefinitionResponse;
(function (GetFunctionDefinitionResponse) {
    GetFunctionDefinitionResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetFunctionDefinitionResponse || (GetFunctionDefinitionResponse = {}));
export var GetFunctionDefinitionVersionRequest;
(function (GetFunctionDefinitionVersionRequest) {
    GetFunctionDefinitionVersionRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetFunctionDefinitionVersionRequest || (GetFunctionDefinitionVersionRequest = {}));
export var GetFunctionDefinitionVersionResponse;
(function (GetFunctionDefinitionVersionResponse) {
    GetFunctionDefinitionVersionResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetFunctionDefinitionVersionResponse || (GetFunctionDefinitionVersionResponse = {}));
export var GetGroupRequest;
(function (GetGroupRequest) {
    GetGroupRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetGroupRequest || (GetGroupRequest = {}));
export var GetGroupResponse;
(function (GetGroupResponse) {
    GetGroupResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetGroupResponse || (GetGroupResponse = {}));
export var GetGroupCertificateAuthorityRequest;
(function (GetGroupCertificateAuthorityRequest) {
    GetGroupCertificateAuthorityRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetGroupCertificateAuthorityRequest || (GetGroupCertificateAuthorityRequest = {}));
export var GetGroupCertificateAuthorityResponse;
(function (GetGroupCertificateAuthorityResponse) {
    GetGroupCertificateAuthorityResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetGroupCertificateAuthorityResponse || (GetGroupCertificateAuthorityResponse = {}));
export var GetGroupCertificateConfigurationRequest;
(function (GetGroupCertificateConfigurationRequest) {
    GetGroupCertificateConfigurationRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetGroupCertificateConfigurationRequest || (GetGroupCertificateConfigurationRequest = {}));
export var GetGroupCertificateConfigurationResponse;
(function (GetGroupCertificateConfigurationResponse) {
    GetGroupCertificateConfigurationResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetGroupCertificateConfigurationResponse || (GetGroupCertificateConfigurationResponse = {}));
export var GetGroupVersionRequest;
(function (GetGroupVersionRequest) {
    GetGroupVersionRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetGroupVersionRequest || (GetGroupVersionRequest = {}));
export var GetGroupVersionResponse;
(function (GetGroupVersionResponse) {
    GetGroupVersionResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetGroupVersionResponse || (GetGroupVersionResponse = {}));
export var GetLoggerDefinitionRequest;
(function (GetLoggerDefinitionRequest) {
    GetLoggerDefinitionRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetLoggerDefinitionRequest || (GetLoggerDefinitionRequest = {}));
export var GetLoggerDefinitionResponse;
(function (GetLoggerDefinitionResponse) {
    GetLoggerDefinitionResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetLoggerDefinitionResponse || (GetLoggerDefinitionResponse = {}));
export var GetLoggerDefinitionVersionRequest;
(function (GetLoggerDefinitionVersionRequest) {
    GetLoggerDefinitionVersionRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetLoggerDefinitionVersionRequest || (GetLoggerDefinitionVersionRequest = {}));
export var GetLoggerDefinitionVersionResponse;
(function (GetLoggerDefinitionVersionResponse) {
    GetLoggerDefinitionVersionResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetLoggerDefinitionVersionResponse || (GetLoggerDefinitionVersionResponse = {}));
export var GetResourceDefinitionRequest;
(function (GetResourceDefinitionRequest) {
    GetResourceDefinitionRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetResourceDefinitionRequest || (GetResourceDefinitionRequest = {}));
export var GetResourceDefinitionResponse;
(function (GetResourceDefinitionResponse) {
    GetResourceDefinitionResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetResourceDefinitionResponse || (GetResourceDefinitionResponse = {}));
export var GetResourceDefinitionVersionRequest;
(function (GetResourceDefinitionVersionRequest) {
    GetResourceDefinitionVersionRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetResourceDefinitionVersionRequest || (GetResourceDefinitionVersionRequest = {}));
export var GetResourceDefinitionVersionResponse;
(function (GetResourceDefinitionVersionResponse) {
    GetResourceDefinitionVersionResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetResourceDefinitionVersionResponse || (GetResourceDefinitionVersionResponse = {}));
export var GetServiceRoleForAccountRequest;
(function (GetServiceRoleForAccountRequest) {
    GetServiceRoleForAccountRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetServiceRoleForAccountRequest || (GetServiceRoleForAccountRequest = {}));
export var GetServiceRoleForAccountResponse;
(function (GetServiceRoleForAccountResponse) {
    GetServiceRoleForAccountResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetServiceRoleForAccountResponse || (GetServiceRoleForAccountResponse = {}));
export var GetSubscriptionDefinitionRequest;
(function (GetSubscriptionDefinitionRequest) {
    GetSubscriptionDefinitionRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetSubscriptionDefinitionRequest || (GetSubscriptionDefinitionRequest = {}));
export var GetSubscriptionDefinitionResponse;
(function (GetSubscriptionDefinitionResponse) {
    GetSubscriptionDefinitionResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetSubscriptionDefinitionResponse || (GetSubscriptionDefinitionResponse = {}));
export var GetSubscriptionDefinitionVersionRequest;
(function (GetSubscriptionDefinitionVersionRequest) {
    GetSubscriptionDefinitionVersionRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetSubscriptionDefinitionVersionRequest || (GetSubscriptionDefinitionVersionRequest = {}));
export var GetSubscriptionDefinitionVersionResponse;
(function (GetSubscriptionDefinitionVersionResponse) {
    GetSubscriptionDefinitionVersionResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetSubscriptionDefinitionVersionResponse || (GetSubscriptionDefinitionVersionResponse = {}));
export var GetThingRuntimeConfigurationRequest;
(function (GetThingRuntimeConfigurationRequest) {
    GetThingRuntimeConfigurationRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetThingRuntimeConfigurationRequest || (GetThingRuntimeConfigurationRequest = {}));
export var Telemetry;
(function (Telemetry) {
    Telemetry["Off"] = "Off";
    Telemetry["On"] = "On";
})(Telemetry || (Telemetry = {}));
export var TelemetryConfiguration;
(function (TelemetryConfiguration) {
    TelemetryConfiguration.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(TelemetryConfiguration || (TelemetryConfiguration = {}));
export var RuntimeConfiguration;
(function (RuntimeConfiguration) {
    RuntimeConfiguration.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(RuntimeConfiguration || (RuntimeConfiguration = {}));
export var GetThingRuntimeConfigurationResponse;
(function (GetThingRuntimeConfigurationResponse) {
    GetThingRuntimeConfigurationResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetThingRuntimeConfigurationResponse || (GetThingRuntimeConfigurationResponse = {}));
export var ListBulkDeploymentDetailedReportsRequest;
(function (ListBulkDeploymentDetailedReportsRequest) {
    ListBulkDeploymentDetailedReportsRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListBulkDeploymentDetailedReportsRequest || (ListBulkDeploymentDetailedReportsRequest = {}));
export var ListBulkDeploymentDetailedReportsResponse;
(function (ListBulkDeploymentDetailedReportsResponse) {
    ListBulkDeploymentDetailedReportsResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListBulkDeploymentDetailedReportsResponse || (ListBulkDeploymentDetailedReportsResponse = {}));
export var ListBulkDeploymentsRequest;
(function (ListBulkDeploymentsRequest) {
    ListBulkDeploymentsRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListBulkDeploymentsRequest || (ListBulkDeploymentsRequest = {}));
export var ListBulkDeploymentsResponse;
(function (ListBulkDeploymentsResponse) {
    ListBulkDeploymentsResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListBulkDeploymentsResponse || (ListBulkDeploymentsResponse = {}));
export var ListConnectorDefinitionsRequest;
(function (ListConnectorDefinitionsRequest) {
    ListConnectorDefinitionsRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListConnectorDefinitionsRequest || (ListConnectorDefinitionsRequest = {}));
export var ListConnectorDefinitionsResponse;
(function (ListConnectorDefinitionsResponse) {
    ListConnectorDefinitionsResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListConnectorDefinitionsResponse || (ListConnectorDefinitionsResponse = {}));
export var ListConnectorDefinitionVersionsRequest;
(function (ListConnectorDefinitionVersionsRequest) {
    ListConnectorDefinitionVersionsRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListConnectorDefinitionVersionsRequest || (ListConnectorDefinitionVersionsRequest = {}));
export var ListConnectorDefinitionVersionsResponse;
(function (ListConnectorDefinitionVersionsResponse) {
    ListConnectorDefinitionVersionsResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListConnectorDefinitionVersionsResponse || (ListConnectorDefinitionVersionsResponse = {}));
export var ListCoreDefinitionsRequest;
(function (ListCoreDefinitionsRequest) {
    ListCoreDefinitionsRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListCoreDefinitionsRequest || (ListCoreDefinitionsRequest = {}));
export var ListCoreDefinitionsResponse;
(function (ListCoreDefinitionsResponse) {
    ListCoreDefinitionsResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListCoreDefinitionsResponse || (ListCoreDefinitionsResponse = {}));
export var ListCoreDefinitionVersionsRequest;
(function (ListCoreDefinitionVersionsRequest) {
    ListCoreDefinitionVersionsRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListCoreDefinitionVersionsRequest || (ListCoreDefinitionVersionsRequest = {}));
export var ListCoreDefinitionVersionsResponse;
(function (ListCoreDefinitionVersionsResponse) {
    ListCoreDefinitionVersionsResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListCoreDefinitionVersionsResponse || (ListCoreDefinitionVersionsResponse = {}));
export var ListDeploymentsRequest;
(function (ListDeploymentsRequest) {
    ListDeploymentsRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListDeploymentsRequest || (ListDeploymentsRequest = {}));
export var ListDeploymentsResponse;
(function (ListDeploymentsResponse) {
    ListDeploymentsResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListDeploymentsResponse || (ListDeploymentsResponse = {}));
export var ListDeviceDefinitionsRequest;
(function (ListDeviceDefinitionsRequest) {
    ListDeviceDefinitionsRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListDeviceDefinitionsRequest || (ListDeviceDefinitionsRequest = {}));
export var ListDeviceDefinitionsResponse;
(function (ListDeviceDefinitionsResponse) {
    ListDeviceDefinitionsResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListDeviceDefinitionsResponse || (ListDeviceDefinitionsResponse = {}));
export var ListDeviceDefinitionVersionsRequest;
(function (ListDeviceDefinitionVersionsRequest) {
    ListDeviceDefinitionVersionsRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListDeviceDefinitionVersionsRequest || (ListDeviceDefinitionVersionsRequest = {}));
export var ListDeviceDefinitionVersionsResponse;
(function (ListDeviceDefinitionVersionsResponse) {
    ListDeviceDefinitionVersionsResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListDeviceDefinitionVersionsResponse || (ListDeviceDefinitionVersionsResponse = {}));
export var ListFunctionDefinitionsRequest;
(function (ListFunctionDefinitionsRequest) {
    ListFunctionDefinitionsRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListFunctionDefinitionsRequest || (ListFunctionDefinitionsRequest = {}));
export var ListFunctionDefinitionsResponse;
(function (ListFunctionDefinitionsResponse) {
    ListFunctionDefinitionsResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListFunctionDefinitionsResponse || (ListFunctionDefinitionsResponse = {}));
export var ListFunctionDefinitionVersionsRequest;
(function (ListFunctionDefinitionVersionsRequest) {
    ListFunctionDefinitionVersionsRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListFunctionDefinitionVersionsRequest || (ListFunctionDefinitionVersionsRequest = {}));
export var ListFunctionDefinitionVersionsResponse;
(function (ListFunctionDefinitionVersionsResponse) {
    ListFunctionDefinitionVersionsResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListFunctionDefinitionVersionsResponse || (ListFunctionDefinitionVersionsResponse = {}));
export var ListGroupCertificateAuthoritiesRequest;
(function (ListGroupCertificateAuthoritiesRequest) {
    ListGroupCertificateAuthoritiesRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListGroupCertificateAuthoritiesRequest || (ListGroupCertificateAuthoritiesRequest = {}));
export var ListGroupCertificateAuthoritiesResponse;
(function (ListGroupCertificateAuthoritiesResponse) {
    ListGroupCertificateAuthoritiesResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListGroupCertificateAuthoritiesResponse || (ListGroupCertificateAuthoritiesResponse = {}));
export var ListGroupsRequest;
(function (ListGroupsRequest) {
    ListGroupsRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListGroupsRequest || (ListGroupsRequest = {}));
export var ListGroupsResponse;
(function (ListGroupsResponse) {
    ListGroupsResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListGroupsResponse || (ListGroupsResponse = {}));
export var ListGroupVersionsRequest;
(function (ListGroupVersionsRequest) {
    ListGroupVersionsRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListGroupVersionsRequest || (ListGroupVersionsRequest = {}));
export var ListGroupVersionsResponse;
(function (ListGroupVersionsResponse) {
    ListGroupVersionsResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListGroupVersionsResponse || (ListGroupVersionsResponse = {}));
export var ListLoggerDefinitionsRequest;
(function (ListLoggerDefinitionsRequest) {
    ListLoggerDefinitionsRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListLoggerDefinitionsRequest || (ListLoggerDefinitionsRequest = {}));
export var ListLoggerDefinitionsResponse;
(function (ListLoggerDefinitionsResponse) {
    ListLoggerDefinitionsResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListLoggerDefinitionsResponse || (ListLoggerDefinitionsResponse = {}));
export var ListLoggerDefinitionVersionsRequest;
(function (ListLoggerDefinitionVersionsRequest) {
    ListLoggerDefinitionVersionsRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListLoggerDefinitionVersionsRequest || (ListLoggerDefinitionVersionsRequest = {}));
export var ListLoggerDefinitionVersionsResponse;
(function (ListLoggerDefinitionVersionsResponse) {
    ListLoggerDefinitionVersionsResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListLoggerDefinitionVersionsResponse || (ListLoggerDefinitionVersionsResponse = {}));
export var ListResourceDefinitionsRequest;
(function (ListResourceDefinitionsRequest) {
    ListResourceDefinitionsRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListResourceDefinitionsRequest || (ListResourceDefinitionsRequest = {}));
export var ListResourceDefinitionsResponse;
(function (ListResourceDefinitionsResponse) {
    ListResourceDefinitionsResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListResourceDefinitionsResponse || (ListResourceDefinitionsResponse = {}));
export var ListResourceDefinitionVersionsRequest;
(function (ListResourceDefinitionVersionsRequest) {
    ListResourceDefinitionVersionsRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListResourceDefinitionVersionsRequest || (ListResourceDefinitionVersionsRequest = {}));
export var ListResourceDefinitionVersionsResponse;
(function (ListResourceDefinitionVersionsResponse) {
    ListResourceDefinitionVersionsResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListResourceDefinitionVersionsResponse || (ListResourceDefinitionVersionsResponse = {}));
export var ListSubscriptionDefinitionsRequest;
(function (ListSubscriptionDefinitionsRequest) {
    ListSubscriptionDefinitionsRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListSubscriptionDefinitionsRequest || (ListSubscriptionDefinitionsRequest = {}));
export var ListSubscriptionDefinitionsResponse;
(function (ListSubscriptionDefinitionsResponse) {
    ListSubscriptionDefinitionsResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListSubscriptionDefinitionsResponse || (ListSubscriptionDefinitionsResponse = {}));
export var ListSubscriptionDefinitionVersionsRequest;
(function (ListSubscriptionDefinitionVersionsRequest) {
    ListSubscriptionDefinitionVersionsRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListSubscriptionDefinitionVersionsRequest || (ListSubscriptionDefinitionVersionsRequest = {}));
export var ListSubscriptionDefinitionVersionsResponse;
(function (ListSubscriptionDefinitionVersionsResponse) {
    ListSubscriptionDefinitionVersionsResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListSubscriptionDefinitionVersionsResponse || (ListSubscriptionDefinitionVersionsResponse = {}));
export var ListTagsForResourceRequest;
(function (ListTagsForResourceRequest) {
    ListTagsForResourceRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListTagsForResourceRequest || (ListTagsForResourceRequest = {}));
export var ListTagsForResourceResponse;
(function (ListTagsForResourceResponse) {
    ListTagsForResourceResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListTagsForResourceResponse || (ListTagsForResourceResponse = {}));
export var ResetDeploymentsRequest;
(function (ResetDeploymentsRequest) {
    ResetDeploymentsRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ResetDeploymentsRequest || (ResetDeploymentsRequest = {}));
export var ResetDeploymentsResponse;
(function (ResetDeploymentsResponse) {
    ResetDeploymentsResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ResetDeploymentsResponse || (ResetDeploymentsResponse = {}));
export var StartBulkDeploymentRequest;
(function (StartBulkDeploymentRequest) {
    StartBulkDeploymentRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(StartBulkDeploymentRequest || (StartBulkDeploymentRequest = {}));
export var StartBulkDeploymentResponse;
(function (StartBulkDeploymentResponse) {
    StartBulkDeploymentResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(StartBulkDeploymentResponse || (StartBulkDeploymentResponse = {}));
export var StopBulkDeploymentRequest;
(function (StopBulkDeploymentRequest) {
    StopBulkDeploymentRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(StopBulkDeploymentRequest || (StopBulkDeploymentRequest = {}));
export var StopBulkDeploymentResponse;
(function (StopBulkDeploymentResponse) {
    StopBulkDeploymentResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(StopBulkDeploymentResponse || (StopBulkDeploymentResponse = {}));
export var TagResourceRequest;
(function (TagResourceRequest) {
    TagResourceRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(TagResourceRequest || (TagResourceRequest = {}));
export var UntagResourceRequest;
(function (UntagResourceRequest) {
    UntagResourceRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UntagResourceRequest || (UntagResourceRequest = {}));
export var UpdateConnectivityInfoRequest;
(function (UpdateConnectivityInfoRequest) {
    UpdateConnectivityInfoRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UpdateConnectivityInfoRequest || (UpdateConnectivityInfoRequest = {}));
export var UpdateConnectivityInfoResponse;
(function (UpdateConnectivityInfoResponse) {
    UpdateConnectivityInfoResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UpdateConnectivityInfoResponse || (UpdateConnectivityInfoResponse = {}));
export var UpdateConnectorDefinitionRequest;
(function (UpdateConnectorDefinitionRequest) {
    UpdateConnectorDefinitionRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UpdateConnectorDefinitionRequest || (UpdateConnectorDefinitionRequest = {}));
export var UpdateConnectorDefinitionResponse;
(function (UpdateConnectorDefinitionResponse) {
    UpdateConnectorDefinitionResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UpdateConnectorDefinitionResponse || (UpdateConnectorDefinitionResponse = {}));
export var UpdateCoreDefinitionRequest;
(function (UpdateCoreDefinitionRequest) {
    UpdateCoreDefinitionRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UpdateCoreDefinitionRequest || (UpdateCoreDefinitionRequest = {}));
export var UpdateCoreDefinitionResponse;
(function (UpdateCoreDefinitionResponse) {
    UpdateCoreDefinitionResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UpdateCoreDefinitionResponse || (UpdateCoreDefinitionResponse = {}));
export var UpdateDeviceDefinitionRequest;
(function (UpdateDeviceDefinitionRequest) {
    UpdateDeviceDefinitionRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UpdateDeviceDefinitionRequest || (UpdateDeviceDefinitionRequest = {}));
export var UpdateDeviceDefinitionResponse;
(function (UpdateDeviceDefinitionResponse) {
    UpdateDeviceDefinitionResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UpdateDeviceDefinitionResponse || (UpdateDeviceDefinitionResponse = {}));
export var UpdateFunctionDefinitionRequest;
(function (UpdateFunctionDefinitionRequest) {
    UpdateFunctionDefinitionRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UpdateFunctionDefinitionRequest || (UpdateFunctionDefinitionRequest = {}));
export var UpdateFunctionDefinitionResponse;
(function (UpdateFunctionDefinitionResponse) {
    UpdateFunctionDefinitionResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UpdateFunctionDefinitionResponse || (UpdateFunctionDefinitionResponse = {}));
export var UpdateGroupRequest;
(function (UpdateGroupRequest) {
    UpdateGroupRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UpdateGroupRequest || (UpdateGroupRequest = {}));
export var UpdateGroupResponse;
(function (UpdateGroupResponse) {
    UpdateGroupResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UpdateGroupResponse || (UpdateGroupResponse = {}));
export var UpdateGroupCertificateConfigurationRequest;
(function (UpdateGroupCertificateConfigurationRequest) {
    UpdateGroupCertificateConfigurationRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UpdateGroupCertificateConfigurationRequest || (UpdateGroupCertificateConfigurationRequest = {}));
export var UpdateGroupCertificateConfigurationResponse;
(function (UpdateGroupCertificateConfigurationResponse) {
    UpdateGroupCertificateConfigurationResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UpdateGroupCertificateConfigurationResponse || (UpdateGroupCertificateConfigurationResponse = {}));
export var UpdateLoggerDefinitionRequest;
(function (UpdateLoggerDefinitionRequest) {
    UpdateLoggerDefinitionRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UpdateLoggerDefinitionRequest || (UpdateLoggerDefinitionRequest = {}));
export var UpdateLoggerDefinitionResponse;
(function (UpdateLoggerDefinitionResponse) {
    UpdateLoggerDefinitionResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UpdateLoggerDefinitionResponse || (UpdateLoggerDefinitionResponse = {}));
export var UpdateResourceDefinitionRequest;
(function (UpdateResourceDefinitionRequest) {
    UpdateResourceDefinitionRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UpdateResourceDefinitionRequest || (UpdateResourceDefinitionRequest = {}));
export var UpdateResourceDefinitionResponse;
(function (UpdateResourceDefinitionResponse) {
    UpdateResourceDefinitionResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UpdateResourceDefinitionResponse || (UpdateResourceDefinitionResponse = {}));
export var UpdateSubscriptionDefinitionRequest;
(function (UpdateSubscriptionDefinitionRequest) {
    UpdateSubscriptionDefinitionRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UpdateSubscriptionDefinitionRequest || (UpdateSubscriptionDefinitionRequest = {}));
export var UpdateSubscriptionDefinitionResponse;
(function (UpdateSubscriptionDefinitionResponse) {
    UpdateSubscriptionDefinitionResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UpdateSubscriptionDefinitionResponse || (UpdateSubscriptionDefinitionResponse = {}));
export var TelemetryConfigurationUpdate;
(function (TelemetryConfigurationUpdate) {
    TelemetryConfigurationUpdate.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(TelemetryConfigurationUpdate || (TelemetryConfigurationUpdate = {}));
export var UpdateThingRuntimeConfigurationRequest;
(function (UpdateThingRuntimeConfigurationRequest) {
    UpdateThingRuntimeConfigurationRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UpdateThingRuntimeConfigurationRequest || (UpdateThingRuntimeConfigurationRequest = {}));
export var UpdateThingRuntimeConfigurationResponse;
(function (UpdateThingRuntimeConfigurationResponse) {
    UpdateThingRuntimeConfigurationResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UpdateThingRuntimeConfigurationResponse || (UpdateThingRuntimeConfigurationResponse = {}));
