// @aws-sdk/client-greengrass/dist/es/commands/DeleteCoreDefinitionCommand.js
import { __extends } from "tslib";
import { DeleteCoreDefinitionRequest, DeleteCoreDefinitionResponse } from "../models/models_0";
import { deserializeAws_restJson1DeleteCoreDefinitionCommand, serializeAws_restJson1DeleteCoreDefinitionCommand, } from "../protocols/Aws_restJson1";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
/**
 * Deletes a core definition.
 */
var DeleteCoreDefinitionCommand = /** @class */ (function (_super) {
    __extends(DeleteCoreDefinitionCommand, _super);
    // Start section: command_properties
    // End section: command_properties
    function DeleteCoreDefinitionCommand(input) {
        var _this = 
        // Start section: command_constructor
        _super.call(this) || this;
        _this.input = input;
        return _this;
        // End section: command_constructor
    }
    /**
     * @internal
     */
    DeleteCoreDefinitionCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "GreengrassClient";
        var commandName = "DeleteCoreDefinitionCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DeleteCoreDefinitionRequest.filterSensitiveLog,
            outputFilterSensitiveLog: DeleteCoreDefinitionResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DeleteCoreDefinitionCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1DeleteCoreDefinitionCommand(input, context);
    };
    DeleteCoreDefinitionCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1DeleteCoreDefinitionCommand(output, context);
    };
    return DeleteCoreDefinitionCommand;
}($Command));
export { DeleteCoreDefinitionCommand };
