// @aws-sdk/client-greengrass/dist/es/commands/DisassociateServiceRoleFromAccountCommand.js
import { __extends } from "tslib";
import { DisassociateServiceRoleFromAccountRequest, DisassociateServiceRoleFromAccountResponse, } from "../models/models_0";
import { deserializeAws_restJson1DisassociateServiceRoleFromAccountCommand, serializeAws_restJson1DisassociateServiceRoleFromAccountCommand, } from "../protocols/Aws_restJson1";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
/**
 * Disassociates the service role from your account. Without a service role, deployments will not work.
 */
var DisassociateServiceRoleFromAccountCommand = /** @class */ (function (_super) {
    __extends(DisassociateServiceRoleFromAccountCommand, _super);
    // Start section: command_properties
    // End section: command_properties
    function DisassociateServiceRoleFromAccountCommand(input) {
        var _this = 
        // Start section: command_constructor
        _super.call(this) || this;
        _this.input = input;
        return _this;
        // End section: command_constructor
    }
    /**
     * @internal
     */
    DisassociateServiceRoleFromAccountCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "GreengrassClient";
        var commandName = "DisassociateServiceRoleFromAccountCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DisassociateServiceRoleFromAccountRequest.filterSensitiveLog,
            outputFilterSensitiveLog: DisassociateServiceRoleFromAccountResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DisassociateServiceRoleFromAccountCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1DisassociateServiceRoleFromAccountCommand(input, context);
    };
    DisassociateServiceRoleFromAccountCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1DisassociateServiceRoleFromAccountCommand(output, context);
    };
    return DisassociateServiceRoleFromAccountCommand;
}($Command));
export { DisassociateServiceRoleFromAccountCommand };
