// @aws-sdk/client-greengrass/dist/es/commands/ListCoreDefinitionVersionsCommand.js
import { __extends } from "tslib";
import { ListCoreDefinitionVersionsRequest, ListCoreDefinitionVersionsResponse } from "../models/models_0";
import { deserializeAws_restJson1ListCoreDefinitionVersionsCommand, serializeAws_restJson1ListCoreDefinitionVersionsCommand, } from "../protocols/Aws_restJson1";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
/**
 * Lists the versions of a core definition.
 */
var ListCoreDefinitionVersionsCommand = /** @class */ (function (_super) {
    __extends(ListCoreDefinitionVersionsCommand, _super);
    // Start section: command_properties
    // End section: command_properties
    function ListCoreDefinitionVersionsCommand(input) {
        var _this = 
        // Start section: command_constructor
        _super.call(this) || this;
        _this.input = input;
        return _this;
        // End section: command_constructor
    }
    /**
     * @internal
     */
    ListCoreDefinitionVersionsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "GreengrassClient";
        var commandName = "ListCoreDefinitionVersionsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListCoreDefinitionVersionsRequest.filterSensitiveLog,
            outputFilterSensitiveLog: ListCoreDefinitionVersionsResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListCoreDefinitionVersionsCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1ListCoreDefinitionVersionsCommand(input, context);
    };
    ListCoreDefinitionVersionsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1ListCoreDefinitionVersionsCommand(output, context);
    };
    return ListCoreDefinitionVersionsCommand;
}($Command));
export { ListCoreDefinitionVersionsCommand };
