// @aws-sdk/client-greengrass/dist/es/commands/ListConnectorDefinitionVersionsCommand.js
import { __extends } from "tslib";
import { ListConnectorDefinitionVersionsRequest, ListConnectorDefinitionVersionsResponse } from "../models/models_0";
import { deserializeAws_restJson1ListConnectorDefinitionVersionsCommand, serializeAws_restJson1ListConnectorDefinitionVersionsCommand, } from "../protocols/Aws_restJson1";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
/**
 * Lists the versions of a connector definition, which are containers for connectors. Connectors run on the Greengrass core and contain built-in integration with local infrastructure, device protocols, AWS, and other cloud services.
 */
var ListConnectorDefinitionVersionsCommand = /** @class */ (function (_super) {
    __extends(ListConnectorDefinitionVersionsCommand, _super);
    // Start section: command_properties
    // End section: command_properties
    function ListConnectorDefinitionVersionsCommand(input) {
        var _this = 
        // Start section: command_constructor
        _super.call(this) || this;
        _this.input = input;
        return _this;
        // End section: command_constructor
    }
    /**
     * @internal
     */
    ListConnectorDefinitionVersionsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "GreengrassClient";
        var commandName = "ListConnectorDefinitionVersionsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListConnectorDefinitionVersionsRequest.filterSensitiveLog,
            outputFilterSensitiveLog: ListConnectorDefinitionVersionsResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListConnectorDefinitionVersionsCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1ListConnectorDefinitionVersionsCommand(input, context);
    };
    ListConnectorDefinitionVersionsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1ListConnectorDefinitionVersionsCommand(output, context);
    };
    return ListConnectorDefinitionVersionsCommand;
}($Command));
export { ListConnectorDefinitionVersionsCommand };
