// @aws-sdk/client-iot/dist/es/commands/CancelJobExecutionCommand.js
import { __extends } from "tslib";
import { CancelJobExecutionRequest } from "../models/models_0";
import { deserializeAws_restJson1CancelJobExecutionCommand, serializeAws_restJson1CancelJobExecutionCommand, } from "../protocols/Aws_restJson1";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
/**
 * <p>Cancels the execution of a job for a given thing.</p>
 */
var CancelJobExecutionCommand = /** @class */ (function (_super) {
    __extends(CancelJobExecutionCommand, _super);
    // Start section: command_properties
    // End section: command_properties
    function CancelJobExecutionCommand(input) {
        var _this = 
        // Start section: command_constructor
        _super.call(this) || this;
        _this.input = input;
        return _this;
        // End section: command_constructor
    }
    /**
     * @internal
     */
    CancelJobExecutionCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IoTClient";
        var commandName = "CancelJobExecutionCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CancelJobExecutionRequest.filterSensitiveLog,
            outputFilterSensitiveLog: function (output) { return output; },
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CancelJobExecutionCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1CancelJobExecutionCommand(input, context);
    };
    CancelJobExecutionCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1CancelJobExecutionCommand(output, context);
    };
    return CancelJobExecutionCommand;
}($Command));
export { CancelJobExecutionCommand };
