// @aws-sdk/client-iot/dist/es/commands/ClearDefaultAuthorizerCommand.js
import { __extends } from "tslib";
import { ClearDefaultAuthorizerRequest, ClearDefaultAuthorizerResponse } from "../models/models_0";
import { deserializeAws_restJson1ClearDefaultAuthorizerCommand, serializeAws_restJson1ClearDefaultAuthorizerCommand, } from "../protocols/Aws_restJson1";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
/**
 * <p>Clears the default authorizer.</p>
 */
var ClearDefaultAuthorizerCommand = /** @class */ (function (_super) {
    __extends(ClearDefaultAuthorizerCommand, _super);
    // Start section: command_properties
    // End section: command_properties
    function ClearDefaultAuthorizerCommand(input) {
        var _this = 
        // Start section: command_constructor
        _super.call(this) || this;
        _this.input = input;
        return _this;
        // End section: command_constructor
    }
    /**
     * @internal
     */
    ClearDefaultAuthorizerCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IoTClient";
        var commandName = "ClearDefaultAuthorizerCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ClearDefaultAuthorizerRequest.filterSensitiveLog,
            outputFilterSensitiveLog: ClearDefaultAuthorizerResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ClearDefaultAuthorizerCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1ClearDefaultAuthorizerCommand(input, context);
    };
    ClearDefaultAuthorizerCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1ClearDefaultAuthorizerCommand(output, context);
    };
    return ClearDefaultAuthorizerCommand;
}($Command));
export { ClearDefaultAuthorizerCommand };
