// @aws-sdk/client-iot/dist/es/commands/ListTargetsForPolicyCommand.js
import { __extends } from "tslib";
import { ListTargetsForPolicyRequest, ListTargetsForPolicyResponse } from "../models/models_1";
import { deserializeAws_restJson1ListTargetsForPolicyCommand, serializeAws_restJson1ListTargetsForPolicyCommand, } from "../protocols/Aws_restJson1";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
/**
 * <p>List targets for the specified policy.</p>
 */
var ListTargetsForPolicyCommand = /** @class */ (function (_super) {
    __extends(ListTargetsForPolicyCommand, _super);
    // Start section: command_properties
    // End section: command_properties
    function ListTargetsForPolicyCommand(input) {
        var _this = 
        // Start section: command_constructor
        _super.call(this) || this;
        _this.input = input;
        return _this;
        // End section: command_constructor
    }
    /**
     * @internal
     */
    ListTargetsForPolicyCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IoTClient";
        var commandName = "ListTargetsForPolicyCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListTargetsForPolicyRequest.filterSensitiveLog,
            outputFilterSensitiveLog: ListTargetsForPolicyResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListTargetsForPolicyCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1ListTargetsForPolicyCommand(input, context);
    };
    ListTargetsForPolicyCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1ListTargetsForPolicyCommand(output, context);
    };
    return ListTargetsForPolicyCommand;
}($Command));
export { ListTargetsForPolicyCommand };
