// @aws-sdk/client-iot/dist/es/commands/ListSecurityProfilesForTargetCommand.js
import { __extends } from "tslib";
import { ListSecurityProfilesForTargetRequest, ListSecurityProfilesForTargetResponse } from "../models/models_1";
import { deserializeAws_restJson1ListSecurityProfilesForTargetCommand, serializeAws_restJson1ListSecurityProfilesForTargetCommand, } from "../protocols/Aws_restJson1";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
/**
 * <p>Lists the Device Defender security profiles attached to a target (thing group).</p>
 */
var ListSecurityProfilesForTargetCommand = /** @class */ (function (_super) {
    __extends(ListSecurityProfilesForTargetCommand, _super);
    // Start section: command_properties
    // End section: command_properties
    function ListSecurityProfilesForTargetCommand(input) {
        var _this = 
        // Start section: command_constructor
        _super.call(this) || this;
        _this.input = input;
        return _this;
        // End section: command_constructor
    }
    /**
     * @internal
     */
    ListSecurityProfilesForTargetCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IoTClient";
        var commandName = "ListSecurityProfilesForTargetCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListSecurityProfilesForTargetRequest.filterSensitiveLog,
            outputFilterSensitiveLog: ListSecurityProfilesForTargetResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListSecurityProfilesForTargetCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1ListSecurityProfilesForTargetCommand(input, context);
    };
    ListSecurityProfilesForTargetCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1ListSecurityProfilesForTargetCommand(output, context);
    };
    return ListSecurityProfilesForTargetCommand;
}($Command));
export { ListSecurityProfilesForTargetCommand };
