// @aws-sdk/client-iot/dist/es/commands/StartThingRegistrationTaskCommand.js
import { __extends } from "tslib";
import { StartThingRegistrationTaskRequest, StartThingRegistrationTaskResponse } from "../models/models_1";
import { deserializeAws_restJson1StartThingRegistrationTaskCommand, serializeAws_restJson1StartThingRegistrationTaskCommand, } from "../protocols/Aws_restJson1";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
/**
 * <p>Creates a bulk thing provisioning task.</p>
 */
var StartThingRegistrationTaskCommand = /** @class */ (function (_super) {
    __extends(StartThingRegistrationTaskCommand, _super);
    // Start section: command_properties
    // End section: command_properties
    function StartThingRegistrationTaskCommand(input) {
        var _this = 
        // Start section: command_constructor
        _super.call(this) || this;
        _this.input = input;
        return _this;
        // End section: command_constructor
    }
    /**
     * @internal
     */
    StartThingRegistrationTaskCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IoTClient";
        var commandName = "StartThingRegistrationTaskCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: StartThingRegistrationTaskRequest.filterSensitiveLog,
            outputFilterSensitiveLog: StartThingRegistrationTaskResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    StartThingRegistrationTaskCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1StartThingRegistrationTaskCommand(input, context);
    };
    StartThingRegistrationTaskCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1StartThingRegistrationTaskCommand(output, context);
    };
    return StartThingRegistrationTaskCommand;
}($Command));
export { StartThingRegistrationTaskCommand };
